apps/admin/src/components/custom/LessonItemsAssigner/LessonItemsAssigner.tsx

```diff
@@ -0,0 +1,387 @@
+      return;
+    }
+
+    setLessonItems((prev) => {
+      const activeLessons = prev.filter(
+        ({ columnId }) => columnId === activeColumnId,
+      );
+      const activeLesson = activeLessons.find(({ id }) => id === activeId);
+      const updatedLesson = prev.find(({ id }) => id === activeLesson?.id);
+
+      if (updatedLesson) {
+        updatedLesson.columnId = overColumnId;
+      }
+
+      return [...prev];
+    });
+  }
+
+  function handleDragEnd(event: DragEndEvent) {
+    const { active, over } = event;
+
+    const activeId = active?.id;
+    const overId = over?.id;
+
+    if (!activeId || !overId) return;
+
+    const activeColumnId = active?.data?.current?.sortable.containerId;
+    const overColumnId = over?.data.current?.sortable?.containerId ?? overId;
+
+    if (!activeColumnId || !overColumnId || activeColumnId !== overColumnId) {
+      return;
+    }
+
+    setLessonItems((prev) => {
+      const activeLessons = prev.filter(
+        ({ columnId }) => columnId === activeColumnId,
+      );
+
+      const overLessons = prev.filter(
+        ({ columnId }) => columnId === overColumnId,
+      );
+
+      const activeLesson = activeLessons.find(({ id }) => id === activeId);
+      const overItem = overLessons.find(({ id }) => id === overId);
+
+      if (!activeLesson) return prev;
+
+      const updatedItems = [
+        ...new Set([
+          ...arrayMove(
+            overLessons,
+            activeLessons.indexOf(activeLesson),
+            overItem ? overLessons.indexOf(overItem) : 0,
+          ),
+          ...prev,
+        ]),
+      ];
+
+      if (
+        activeColumnId === "column-assigned" ||
+        overColumnId === "column-assigned"
+      ) {
+        (async () => {
+          await handleChangeLessorOrder(updatedItems, activeLesson);
+
+          if (!activeLesson.params.lesson_item_id) {
+            await handleLessonItemAssign(
+              activeLesson,
+              updatedItems.indexOf(activeLesson),
+            );
+          }
+        })();
+      }
+
+      if (
+        activeColumnId === "column-unassigned" ||
+        overColumnId === "column-unassigned"
+      ) {
+        handleRemove(activeLesson);
+      }
+
+      return updatedItems;
+    });
+
+    setCurrentlyDraggedItem(null);
+  }
+
+  useEffect(() => {
+    fetchLessonItems();
+  }, [lessonId]);
+
+  return (
+    <Box
+      grid
+      style={{
+        display: "grid",
+        gridTemplateColumns: "repeat(2, minmax(0, 1fr))",
+        width: "100%",
+        gap: "1.5rem",
+      }}
+    >
+      <DndContext
+        sensors={sensors}
+        collisionDetection={closestCorners}
+        onDragStart={handleDragStart}
+        onDragOver={handleDragOver}
+        onDragEnd={handleDragEnd}
+      >
+        <DragOverlay>
+          {currentlyDraggedItem ? (
+            <LessonItemCard lessonItem={currentlyDraggedItem} />
+          ) : null}
```

Comment:
as we dont have the second case when `currentlyDraggedItem` is falsy, what do you think about this?
```ts
{
  currentlyDraggedItem && <LessonItemCard lessonItem={currentlyDraggedItem} />
}
```